k9mail/src/main/java/com/fsck/k9/helper/Contacts.java

```diff
@@ -142,6 +142,26 @@ public boolean isInContacts(final String emailAddress) {
     }
 
     /**
+     * Check whether one of the provided addresses belongs to one of the contacts.
+     *
+     * @param addresses The addresses to search in contacts
+     * @return <tt>true</tt>, if one address belongs to a contact.
+     *         <tt>false</tt>, otherwise.
+     */
+    public boolean containsContact(final Address[] addresses) {
```

Comment:
+1 on the name.

Right now we use a "lookup URI" for the search. I'm not sure if the contacts content provider supports arbitrary selection arguments. In this case we should be fine since most emails only have one sender address and the method is called from a background thread during message sync.